types/rest/rest.go

```diff
@@ -252,29 +258,25 @@ func PostProcessResponse(w http.ResponseWriter, cliCtx context.CLIContext, respo
 		}
 	}
 
-	// inject the height into the response by:
-	// - decoding into a map
-	// - adding the height to the map
-	// - encoding using standard JSON library
-	if cliCtx.Height > 0 {
-		m := make(map[string]interface{})
-		err := json.Unmarshal(output, &m)
-		if err != nil {
-			WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-			return
-		}
+	wrappedResp := ResponseWithHeight{
+		Height: cliCtx.Height,
+		Result: result,
+	}
 
-		m["height"] = cliCtx.Height
+	var (
+		output []byte
+		err    error
+	)
 
-		if cliCtx.Indent {
-			output, err = json.MarshalIndent(m, "", "  ")
```

Comment:
we can probably switch back to using `cliCtx.Codec` as opposed to `json`

Reply:
Indeed